Casks/w/wiso-steuer-2022.rb

```diff
@@ -1,23 +1,30 @@
 cask "wiso-steuer-2022" do
   # NOTE: "2022" is not a version number, but an intrinsic part of the product name
-  version "29.15.4410"
+  version "29.15.4410-HF1"
   sha256 "4220eb5630481e2634cc571bcf806196445cb2b3b07db988273ea94aeac90563"
 
-  url "https://update.buhl-data.com/Updates/Steuer/2022/Mac/Files/#{version}-HF1/SteuerMac2022-#{version}.dmg",
+  url "https://update.buhl-data.com/Updates/Steuer/2022/Mac/Files/#{version}/SteuerMac2022-#{version.split("-").first}.dmg",
       verified: "update.buhl-data.com/Updates/Steuer/"
   name "WISO Steuer 2022"
   desc "Tax declaration for the fiscal year 2021"
   homepage "https://www.buhl.de/download/wiso-steuer-2022/"
 
   livecheck do
     url "https://update.buhl-data.com/Updates/Steuer/2022/Mac/Aktuell/appcast-steuer.xml"
-    strategy :sparkle
+    regex(%r{/v?(\d+(?:\.\d+)+[^/]*)/SteuerMac2022[._-]v?(\d+(?:\.\d+)+)\.dmg}i)
+    strategy :sparkle do |item, regex|
+      match = item.url&.match(regex)
+      next if match.blank?
+
+      match[1]
+    end
   end
 
   auto_updates true
   depends_on macos: ">= :mojave"
 
-  app "SteuerMac 2022.app"
+  # Renamed for consistency: app name differs in Finder to shell
+  app "SteuerMac 2022.app", target: "WISO Steuer-Mac 2022.app"
```

Comment:
```suggestion
  app "SteuerMac 2022.app", target: "WISO Steuer 2022.app"
```

We probably should remove the `-Mac` part to be consistent with the other cask (`WISO Steuer 2023.app` and `WISO Steuer 2024.app`).

Reply:
I downloaded the dmg file and compared the app names and it's "WISO Steuer-Mac 2022.app" for 2022. Starting from 2023 the app name changed to "WISO Steuer 2023.app".
What way do you prefer? Consistend names vs. the app names, how they are provided in the dmg files.

Reply:
On second thought, you are right. It should be consistent with what is shown in the Finder (`WISO Steuer-Mac 2022.app`).